app/roles/planned_timetable_slot.rb

```diff
@@ -61,16 +61,12 @@ def generate_teaching_times(weeks_dates)
     if weeks
       #  timetable customized for different weeks defined by string
 
-      week_intervals.map do |interval|
-        if interval.length > 2
-          # true interval
-          interval_split = interval.split('-')
-          Sirius::TeachingTime.new(teaching_period: teaching_period, day: day, start_date: weeks_dates[Integer(interval_split[0]) - 1][0], end_date: weeks_dates[Integer(interval_split[1]) - 1][1])
-        else
-          # one week
-          week_number = Integer(interval) - 1
-          Sirius::TeachingTime.new(teaching_period: teaching_period, day: day, start_date: weeks_dates[week_number][0], end_date: weeks_dates[week_number][1])
-        end
+      weeks.map do |interval|
+        range = interval.to_range
+        from = range.begin
+        to = range.last(1)[0]
```

Comment:
Better use [Range#end](https://ruby-doc.org/core-2.5.1/Range.html#method-i-end).

Reply:
`Range.end` does not work here. Postgres converts automatically all ranges to format `[1,2)` - with end element excluded. Sequel then converts these ranges to ruby's `Range` with end excluded. 

`Range.end` returns the last element of the `Range` event if it is excluded. So there are probably two options, use `.last(1)` or convert Range with end excluded `[start, end)` to Range with end included `[start, end -1]`.

Reply:
@jirutka Which one do you prefer? Or is there a better option?

Reply:
Aha, I didn’t realize that. Well, then you can use `from, to = range.minmax`.

Reply:
yes, it is much better